File: VBLib_Common/Model/Person.vb

Imports System.Collections.Generic
Imports System.ComponentModel.DataAnnotations.Schema

Namespace Model
    Public Class Person
        <DatabaseGenerated(DatabaseGeneratedOption.Identity)>
        Public Property Id As Integer
        Public Property Name As String
        <InverseProperty("LovedBy")>
        Public Overridable Property AnimalsLoved As List(Of Animal)
        <InverseProperty("HatedBy")>
        Public Overridable Property AnimalsHated As List(Of Animal)

        Public Overrides Function ToString() As String
            Return $"Person : Id = {Id} Name = {Name}"
        End Function
    End Class
End Namespace

File: FrameworkLib_Common/Model/Person.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;

namespace FrameworkLib_Common.Model
{
    public class Person
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        public string Name { get; set; } = null!;

        [InverseProperty("LovedBy")]
        public virtual List<Animal>? AnimalsLoved { get; set; }

        [InverseProperty("HatedBy")]
        public virtual List<Animal>? AnimalsHated { get; set; }

        public override string ToString()
        {
            return $"Person : Id = {Id} Name = {Name}";
        }
    }
}